src/main/webapp/src/redux/socket/index.js

```diff
@@ -1,18 +1,35 @@
 // @flow
+import getStore from '../getStore'
+
 import getConfig from 'next/config'
 
 const { publicRuntimeConfig } = getConfig()
 
-const createSocket = () => {
-  const { socketUrl } = publicRuntimeConfig
-  return new WebSocket(`${socketUrl}/sockets`)
+export class Socket {
+  +socket: WebSocket
+
+  constructor() {
+    const { socketUrl } = publicRuntimeConfig
+    this.socket = new WebSocket(`${socketUrl}/sockets`)
+  }
+
+  open = () => {
+    this.socket.onmessage = (event: MessageEvent) => {
```

Comment:
I suggest to also provide a handler for the other events: https://developer.mozilla.org/en-US/docs/Web/API/WebSocket

onerror and onclose should probably be also used so that we can react  in the most reasonable way.

Reply:
Hmm.. Yes, but this needs some error handling code in `redux`. It's maybe too early now. Should we just create (another lol) issue? ;)